docs/source/how_to_guides/cloud_providers.md

```diff
@@ -0,0 +1,80 @@
+# Cloud provider
+
+Streaming dataset supports the following cloud storage providers: [AWS S3](https://aws.amazon.com/s3/), [GCP Storage](https://cloud.google.com/storage), and [OCI Cloud Storage](https://www.oracle.com/cloud/storage/) to stream your data directly to your instance. For [MosaicML platform](https://www.mosaicml.com/cloud) users, follow the steps mentioned in this [AWS S3](https://mcli.docs.mosaicml.com/en/latest/secrets/s3.html), [GCP Storage](https://mcli.docs.mosaicml.com/en/latest/secrets/gcp.html), and [OCI Cloud Storage](https://mcli.docs.mosaicml.com/en/latest/secrets/oci.html) doc on how to configure the cloud provider credentials.
+
+## AWS S3
+
+For S3 bucket with a public access, skip rest of the below steps since no credentials setup is required.
+
+First, make sure the `awscli` is installed, and then run `aws configure` to create the config and credential files:
+
+```
+python -m pip install awscli
+aws configure
+```
+
+Note: the requested credentials can be retrieved through your [AWS console](https://aws.amazon.com/console/), typically under “Command line or programmatic access”.
+
+Your config and credentials files should follow the standard structure output by `aws configure`:
+
+`~/.aws/config`
+
+```
+[default]
+region=us-west-2
+output=json
+
+```
+
+`~/.aws/credentials`
+
+```
+[default]
+aws_access_key_id=AKIAIOSFODNN7EXAMPLE
+aws_secret_access_key=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
+
+```
+
+More details on these files can be found [here](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html). The streaming dataset reads the `~/.aws/config` and `~/.aws/credentials` to authenticate your credentials and stream data into your instance.
+
+## GCP
+
+Streaming dataset support [GCP user credentials](https://cloud.google.com/storage/docs/authentication#user_accounts), where you need to set your GCP user access key and GCP user access secret as environment variables for your runs. You can set these environment variables in your main python script as such.
+
+```python
+import os
+os.environ['GCS_KEY'] = 'AKIAIOSFODNN7EXAMPLE'
+os.environ['GCS_SECRET'] = 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY'
+```
+
+Or you can also export the environment from the shell command
```

Comment:
```suggestion
Alternatively via shell configuration:
```

Reply:
I actually think it's possible to combine the code examples above into a single cell in Sphinx so users just select a single tab to see the Python vs Bash examples.

Reply:
Sphinx tabs is what you want!
https://sphinx-tabs.readthedocs.io/en/latest/

Reply:
This is a great suggestion. Addressed in https://github.com/mosaicml/streaming/pull/169/commits/018e7db3bb1bd25ef935e5916d831307b33033f0.